index.js

```diff
@@ -98,43 +98,58 @@ function Hook (modules, options, onrequire) {
     } else {
       const stat = parse(filename)
       if (stat === undefined) {
-        debug('could not parse filename: %s', filename)
-        return exports // abort if filename could not be parsed
-      }
-      moduleName = stat.name
-      basedir = stat.basedir
-
-      const fullModuleName = resolveModuleName(stat)
-
-      debug('resolved filename to module: %s (id: %s, resolved: %s, basedir: %s)', moduleName, id, fullModuleName, basedir)
-
-      // Ex: require('foo/lib/../bar.js')
-      // moduleName = 'foo'
-      // fullModuleName = 'foo/bar'
-      if (hasWhitelist === true && modules.includes(moduleName) === false) {
-        if (modules.includes(fullModuleName) === false) return exports // abort if module name isn't on whitelist
-
-        // if we get to this point, it means that we're requiring a whitelisted sub-module
-        moduleName = fullModuleName
-      } else {
-        // figure out if this is the main module file, or a file inside the module
-        let res
-        try {
-          res = resolve.sync(moduleName, { basedir })
-        } catch (e) {
-          debug('could not resolve module: %s', moduleName)
-          return exports // abort if module could not be resolved (e.g. no main in package.json and no index.js file)
+        if (path.isAbsolute(filename) && hasWhitelist === true) {
+          const parsedPath = path.parse(filename)
+          if (modules.includes(filename) || modules.includes(path.join(parsedPath.root, parsedPath.dir, parsedPath.name))) {
```

Comment:
What edge-case does the 2nd `modules.includes()` cover? I notice that it doesn't include the file-extension (I'm not sure if this is on purpose).

Reply:
If filename is `'/path/to/file.js'`, this turns into `modules.includes('/path/to/file.js') || modules.includes('/path/to/file')`.

It allows the user to specify or not the `.js` file extension

Reply:
Added comments to clarify

Reply:
Ah ok, it's to allow the user to get the same experience no matter if they call `hook(['/a'], ...)` or `hook(['/a.js'], ...)`.

I had a couple of thoughts while thinking about this, all based on the following:

> `filename` is generated by `Module._resolveFilename(id, this)`, which should normalize all paths, so that both `/foo/bar/index` and `/foo/bar` becomes `/foo/bar/index.js` and that relative paths like `./foo` becomes `/path/to/foo.js` etc.

Thoughts:

- The `path.isAbolute(filename)` check above will always return `true` and can therefore be omitted.
- Since `filename` is always an absolute path, I think it's best if we made support for absolute paths something the user had to actively turn on using a config option. Otherwise one could argue if this was a breaking change. So the `path.isAbolute(filename)` check could be replaced with a check for this config option instead.
- Nit: I think it's a reasonable requirement to ask the user to include the file extension in absolute paths when calling `hook()`. That way we could do away with the 2nd call to `modules.include()`. Would this be annoying for the user you think?

Reply:
@dyladan any thoughts on this?

Reply:
Oops I totally missed this.

>The path.isAbolute(filename) check above will always return true and can therefore be omitted.

Will remove

> Since filename is always an absolute path, I think it's best if we made support for absolute paths something the user had to actively turn on using a config option. Otherwise one could argue if this was a breaking change. So the path.isAbolute(filename) check could be replaced with a check for this config option instead.

If filename is always an absolute path, `modules.includes(filename)` will only be true if the user included an absolute path in the modules array. I think including an absolute path in the array sufficiently conveys the intent of the user that they would like to hook absolute paths, and I think a config option would be redundant. Unless I'm missing some use case where that assumption is incorrect?

>Nit: I think it's a reasonable requirement to ask the user to include the file extension in absolute paths when calling hook(). That way we could do away with the 2nd call to modules.include(). Would this be annoying for the user you think

I'm fine with making it so that absolute paths have to be truly absolute, including the file extension. This makes the change a little safer as well.